proposals/bot-controller-slingshot.md

```diff
@@ -0,0 +1,165 @@
+   - a failure status for the retrieval (along with the exact error message of the failure)
+  
+  Here is an example of what the metadata file might look like:
+  
+```
+[
+  // example of a successful retrieval
+  {
+     "request":{"datetime":"YYYY-MM-DD hh:mm:ss","minerId:"f0XXXX","CID":"bafxxxxxxxxxxxxxxxxxxxxxxxxxxx"}, // <-- the input dealbot team was given
+     "retrievalResult":"success",
+     "ipfsHash":"Qmxxxxxxxxxxxxxxxxxxxxxxxxxxx", // <-- where the retrieved data lives on Riba's ipfs node
+  },
+  
+  // example of a failed retrieval
+  {
+     "request":{"datetime":"YYYY-MM-DD hh:mm:ss","minerId:"f0YYYY","CID":"bafyyyyyyyyyyyyyyyyyyyyyyyyyyy"},
+     "retrievalResult":"failure",
+     "errorMessage":"deal failed: (State=26) error calling node: publishing deal: mpool push: failed to push message: not enough funds (required: 0.19999999998338535 FIL, balance: 0.037461935577982576 FIL): not enough funds to execute transaction",
+  },
+  
+  ...more retrieval attempt objects...
+]
+```
+
+#### Impact
+_How would this directly contribute to web3 dev stack product-market fit?_
+
+<!--
+Explain how this addresses known challenges or opportunities.
+What awesome potential impact/outcomes/results will we see if we nail this project?
+-->
+
+This work enables the Slingshot competition, which increases usage of our stack by onboarding more useful content onto the Filecoin network.
+
+#### Internal leverage
+_How much would nailing this project improve our knowledge and ability to execute future projects?_
+
+<!--
+Explain the opportunity or leverage point for our subsequent velocity/impact (e.g. by speeding up development, enabling more contributors, etc)
+-->
+
+ - There is the potential for cross-learning between this project and other controllers (like [#87](https://github.com/protocol/web3-dev-team/pull/87))
+ - There is the potential to find mainnet bugs in the Dealbot through this project, since it implements a unique type of controller that retrieves data not stored by the Dealbot itself.
+
+#### Confidence
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+<!--Explain why this rating-->
+
+Confidence:  High
+
+This controller utilizes only a subset of the dealbot functionality (only retrieval deals), so if we can get [#87](https://github.com/protocol/web3-dev-team/pull/87) to work, this project should be no more difficult to make work as well.
+
+
+## Project definition
+#### Brief plan of attack
+
+<!--Briefly describe the milestones/steps/work needed for this project-->
+
+#### What does done look like?
+_What specific deliverables should completed to consider this project done?_
+
+ - The go-ipfs node (to be set up by @ribasushi outside this project proposal) should contain the retrieved bits of all successful retrieval
```

Comment:
this seems separate from the controller work, though?

Reply:
Yes, it is separate from the controller.  I just wanted to capture it here because it'll mean small config changes in the Lotus instance used in the controller project.